src/runtime/schema/schema.ts

```diff
@@ -66,18 +81,35 @@ export function create(): SchemaInternal {
     contextContributors: [],
   }
 
+  const middleware: SchemaInternal['public']['middleware'] = (fn) => {
+    api.public.use(
```

Comment:
One hand, cool re-use/dog-fooding. Also though wonder if we'll run into trouble later on. For example framework debug/trace logs to show users what their app is doing in detail, getting confused by this dogfooding.

Just food for thought, no issue right now.